File: CardIndex/DAL/Entities/Article.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DAL.Entities
{
    public class Article
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
        public string AuthorFullName { get; set; }
        public DateTime Created { get; set; }
        [ForeignKey(nameof(Theme))]
        public int ThemeId { get; set; }

        public virtual Theme Theme { get; set; }
        public virtual ICollection<ArticleRate> ArticleRates { get; set; }

    }
}

File: CardIndex/CardIndex/ClientApp/src/app/models/article.ts
export class Article{
    id: number
    title: string
    body: string
    authorFullName: string
    themeName: string
    Created: Date
    AvgRate: number

    constructor(id: number,
        title: string,
        body: string,
        authorFullName: string,
        themeName: string,
        Created: Date,
        AvgRate: number)
        {
            this.id = id;
            this.title = title;
            this.body = body;
            this.authorFullName = authorFullName;
            this.themeName = themeName;
            this.Created = Created;
            this.AvgRate = AvgRate;
        }
}